From 2dcc7b39462a866fcf7b0ab19c97c3910ea45eae Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Fri, 25 Oct 2019 10:48:55 -0700
Subject: [PATCH 575/830] ANDROID: gki_defconfig: add FORTIFY_SOURCE, remove
 SPMI_MSM_PMIC_ARB
X-NVConfidentiality: public

Force SPMI_MSM_PMIC_ARB off and enable FORTIFY_SOURCE for GKI

Bug: 142896904
Bug: 134087016
Test: Build/Boot on cuttlefish
Change-Id: Ib6e501f1efee2349d750010a53572dddcb5457c5
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 arch/arm64/configs/gki_defconfig | 2 ++
 arch/x86/configs/gki_defconfig   | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 56c624067954..33b742543ec4 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -284,6 +284,7 @@ CONFIG_HW_RANDOM_VIRTIO=y
 # CONFIG_I2C_HELPER_AUTO is not set
 CONFIG_SPI=y
 CONFIG_SPMI=y
+# CONFIG_SPMI_MSM_PMIC_ARB is not set
 CONFIG_PINCTRL_AMD=y
 CONFIG_POWER_RESET_HISI=y
 # CONFIG_HWMON is not set
@@ -446,6 +447,7 @@ CONFIG_NLS_UTF8=y
 CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 CONFIG_HARDENED_USERCOPY=y
+CONFIG_FORTIFY_SOURCE=y
 CONFIG_SECURITY_SELINUX=y
 CONFIG_CRYPTO_ADIANTUM=y
 CONFIG_CRYPTO_MD4=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index 5add23fc3d0b..e19e3343d322 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -382,6 +382,7 @@ CONFIG_NLS_UTF8=y
 CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 CONFIG_HARDENED_USERCOPY=y
+CONFIG_FORTIFY_SOURCE=y
 CONFIG_SECURITY_SELINUX=y
 CONFIG_CRYPTO_ADIANTUM=y
 CONFIG_CRYPTO_SHA256_SSSE3=y
-- 
2.17.1

